store/daystore/day_store.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package store
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+	"time"
+
+	statstore "k8s.io/heapster/store/statstore"
+
+	"k8s.io/heapster/third_party/window"
+)
+
+// DayStore is an in-memory window of derived stats for each hour of a day.
+// DayStore holds 24 hours of derived stats, plus an hour-long StatStore that holds -
+// historical data on the current hour.
+// DayStore can calculate the Average, Max and 95th Percentile over the past 24 hours.
+// The latest one hour is only considered for the calculation of the daily Max.
```

Comment:
I guess the comment is misleading. What I meant to say is the following (will be edited in a new commit):

if the DayStore has data within the past 0-1 hour:
- The Average, Max and 95th pecentile are the corresponding values of the underlying StatStore

if the DayStore has more than 1 hour of data then hourly stats have been flushed to the window buffer. Then:
- The Max is calculated as the max of all entries in the window, and the max of the underlying StatStore
- The Average and 95th percentiles are calculated from the entries in the window. the values in the underlying StatStore are NOT considered as some datapoints will be accounted for twice. For example, if a flush happened 10 minutes ago, the Hour StatStore will hold 50 minutes of data that have been accounted for in the latest hourEntry of the window. 

If this is a concern on the accuracy of daily stats, I could Get all the TimePoints from the Hour StatStore that are later than the last hourly flush, and weigh these points appropriately for the calculation of the average. For the 95th percentile, we could calculate the 95th percentile of the remainder of the past one hour and use that to calculate the daily 95th percentile. WDYT?